Bolt Card Component Refactor

mikemai2awesome opened the pull request:
## Jira

http://vjira2:8080/browse/BDS-685

## Summary

Enhanced card component to be easily accept free-form content and added whole card clickability feature.

## Details
* Refactored twig templates and CSS to accommodate free-form content
* Web component added
* Whole card clickability is implemented

## How to test

Run the branch locally and navigate to the Card docs page, make sure all demos are doing what they are supposed to do.

mikemai2awesome commented:
@remydenton @sghoweri 

The last holdup on this is that `bolt-card-link` cannot have an empty `href` when it is not a link but a function, such as play/pause the video. If `href` is empty, it will jump the page.

danielamorse commented on packages/components/bolt-card/src/_card-link.js:
```diff
@@ -0,0 +1,36 @@
+let cx = classNames.bind(styles);
+
+@define
+class BoltCardLink extends BoltLink {
```
@mikemai2awesome @sghoweri I think we can rework this to extend `BoltAction` now.

danielamorse replied:
Also noticed that external links aren't opening in new tab, [see example](https://feature-bds-134-card-refactor.boltdesignsystem.com/pattern-lab/patterns/02-components-card-10-card-button-variations/02-components-card-10-card-button-variations.html).

mikemai2awesome replied:
@danielamorse @sghoweri Looks like it's an issue with bolt-button not passing the `target`, maybe it'd be fixed if I pull in master?

danielamorse replied:
@mikemai2awesome reviewed again now that Link updates have been merged in. I'm seeing issues in IE11:
- Video start/stop is broken (when triggered by button or whole-card clink)
- Card link text is not visually hidden
- Card links do not work on web component cards.

I'm going to debug these issues myself since I worked on them before.

sghoweri commented:
@mikemai2awesome @adamszalapski @margoromo just confirming: my feedback from February 15 hasn’t had any progress recently, right?

adamszalapski commented:
@sghoweri not from my side. I want to work on it but we agree that is not a priority at that time.

mikemai2awesome commented:
@sghoweri Not touched yet. I can get on it next sprint since it'll be a quiet week.

mikemai2awesome commented:
@sghoweri 

I've pulled in master and resolve any conflicts. I addressed all your Now feedback. I couldn't get your code to work, so I updated the adaptor twig the best I can. Deprecated cards work without js now.

I also addressed Soon 5 and 7.

sghoweri commented:
Closing out in lieu of https://github.com/boltdesignsystem/bolt/pull/1649 -- which includes all of the initial work from this PR, adds a handful of improvements for the initial release, and gets the ball rolling on some additional improvements to roll out in a future release.
